File: app/src/main/java/com/kit/projectdesign/net/ApiClient.java
package com.kit.projectdesign.net;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiClient {
    public static EntriesApi create(String baseUrl, AuthInterceptor.TokenProvider provider){
        HttpLoggingInterceptor log = new HttpLoggingInterceptor();
        log.setLevel(HttpLoggingInterceptor.Level.BODY);
        OkHttpClient client = new OkHttpClient.Builder()
                .addInterceptor(new AuthInterceptor(provider))
                .addInterceptor(log)
                .build();
        Gson gson = new GsonBuilder().create();
        return new Retrofit.Builder()
                .baseUrl(baseUrl)
                .client(client)
                .addConverterFactory(GsonConverterFactory.create(gson))
                .build()
                .create(EntriesApi.class);
    }
}

File: app/src/main/java/com/kit/projectdesign/api/ApiClient.kt
package com.kit.projectdesign.api

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object ApiClient {
    // 生产环境 URL
    private const val BASE_URL = "http://pd.yagu1125.com/" 
    
    // 开发环境 (本地测试) 请使用: "http://10.0.2.2:3031/"

    private val retrofit by lazy {
        Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }

    val service: ApiService by lazy {
        retrofit.create(ApiService::class.java)
    }
}
